src/stack/stack.jl

```diff
@@ -195,8 +202,12 @@ function Base.merge(
     merge(map(layers, (x1, x2, x3, xs...))...)
 end
 
-Base.map(f, s::AbstractDimStack) = _maybestack(s,map(f, values(s)))
-function Base.map(
+Base.map(f, s::AbstractDimStack) = error("Use maplayers(f, stack)) instad of map(f, stack)")
+Base.map(f, ::Union{AbstractDimStack,NamedTuple}, xs::Union{AbstractDimStack,NamedTuple}...) =
+    error("Use maplayers(f, stack, args...)) instad of map(f, stack, args...)")
```

Comment:
I guess this is the cost...but should it be?  Arguably if you expected `iterate` to work the way it will now, you also expected `map` to work that way...

Is it a common usage pattern to do eg `map(stack) do array`?

Reply:
Yeah, I've meant to deprecate this for ages anyway. The worse cost is how different splatting to a Tuple and a NamedTuple is.

Its a common pattern, but `maplayers` should be fine to use for it.